src/test/DssProxyActionsCharter.t.sol

```diff
@@ -0,0 +1,596 @@
+    }
+
+    function wipe(address, bytes32, address, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function wipeAll(address, bytes32, address) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function lockETHAndDraw(address, address, address, address, uint256) public payable {
+        (bool success,) = address(proxy).call{value: msg.value}(
+            abi.encodeWithSignature("execute(address,bytes)", dssProxyActions, msg.data)
+        );
+        require(success, "");
+    }
+
+    function lockGemAndDraw(address, address, address, address, uint256, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function wipeAndFreeETH(address, address, address, uint256, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function wipeAllAndFreeETH(address, address, address, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function wipeAndFreeGem(address, address, address, uint256, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function wipeAllAndFreeGem(address, address, address, uint256) public {
+        proxy.execute(dssProxyActions, msg.data);
+    }
+
+    function end_freeETH(address a, address b, address c) public {
+        proxy.execute(dssProxyActionsEnd, abi.encodeWithSignature("freeETH(address,address,address)", a, b, c));
+    }
+
+    function end_freeGem(address a, address b, address c) public {
+        proxy.execute(dssProxyActionsEnd, abi.encodeWithSignature("freeGem(address,address,address)", a, b, c));
+    }
+
+    function end_pack(address a, address b, uint256 c) public {
+        proxy.execute(dssProxyActionsEnd, abi.encodeWithSignature("pack(address,address,uint256)", a, b, c));
+    }
+
+    function end_cashETH(address a, address b, address c, bytes32 d, uint256 e) public {
+        proxy.execute(dssProxyActionsEnd, abi.encodeWithSignature("cashETH(address,address,address,bytes32,uint256)", a, b, c, d, e));
+    }
+
+    function end_cashGem(address a, address b, address c, bytes32 d, uint256 e) public {
+        proxy.execute(dssProxyActionsEnd, abi.encodeWithSignature("cashGem(address,address,address,bytes32,uint256)", a, b, c, d, e));
+    }
+}
+
+contract DssProxyActionsTest is DssDeployTestBase, ProxyCalls {
+    CharterManagerImp charter;
+    address charterProxy;
+
+    ManagedGemJoin ethManagedJoin;
+    ManagedGemJoin wbtcJoin;
+    WBTC wbtc;
+    DSValue pipWBTC;
+    ProxyRegistry registry;
+    WETH9_ realWeth;
+
+    function cheat_cage() public {
+        HevmStoreLike(address(hevm)).store(address(vat), bytes32(uint256(10)), bytes32(uint256(0)));
+    }
+
+    function setUp() public override {
+        super.setUp();
+        deployKeepAuth();
+
+        // Create a real WETH token and replace it with a new adapter in the vat
+        realWeth = new WETH9_();
+        this.deny(address(vat), address(ethManagedJoin));
```

Comment:
This should still be `this.deny(address(vat), address(ethJoin));` as this will `deny` the original join created in `DssDeployTestBase`

Reply:
Done